include/pybind11/detail/internals.h

```diff
@@ -34,7 +36,10 @@
 /// further ABI-incompatible changes may be made before the ABI is officially
 /// changed to the new version.
 #ifndef PYBIND11_INTERNALS_VERSION
-#    if PY_VERSION_HEX >= 0x030C0000 || defined(_MSC_VER)
+#    if PY_VERSION_HEX >= 0x030D0000
+// Version bump for Python 3.13+.
```

Comment:
Rather than making this a purely version-specific thing, I would rather append something to the version identifier when performing a build for the free-threaded version.